x-pack/plugins/security_solution/public/detections/components/rules/step_rule_actions/index.tsx

```diff
@@ -139,22 +146,79 @@ const StepRuleActionsComponent: FC<StepRuleActionsProps> = ({
     </>
   );
 };
+
 export const StepRuleActions = memo(StepRuleActionsComponent);
 
 const StepRuleActionsReadOnlyComponent: FC<StepRuleActionsReadOnlyProps> = ({
   addPadding,
-  defaultValues: data,
+  defaultValues: ruleActionsData,
 }) => {
   const {
-    services: {
-      triggersActionsUi: { actionTypeRegistry },
-    },
+    services: { triggersActionsUi },
   } = useKibana();
-  const schema = useMemo(() => getSchema({ actionTypeRegistry }), [actionTypeRegistry]);
+
+  const actionTypeRegistry = triggersActionsUi.actionTypeRegistry as ActionTypeRegistryContract;
+
+  const { data: connectors } = useFetchConnectors();
+  const { data: connectorTypes } = useFetchConnectorTypes();
+
+  const notificationActions = ruleActionsData.actions;
+  const responseActions = ruleActionsData.responseActions || [];
+
+  const ruleHasNoActions = notificationActions.length === 0 && responseActions.length === 0;
+
+  if (ruleHasNoActions || !connectors || !connectorTypes || !ruleActionsData) {
```

Comment:
Probably the problem is here that we do not show panel if there are no connectors, but repsonseActions exist?

Reply:
Great catch! Thanks. Pushed the fix.